app/components-react/shared/Display.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useRef } from 'react';
+import { useVuex } from '../hooks';
+import { Services } from '../service-provider';
+import { Display as OBSDisplay } from '../../services/video';
+import uuid from 'uuid/v4';
+
+interface DisplayProps {
+  sourceId?: string;
+  paddingSize?: number;
+  drawUI?: boolean;
+  renderingMode?: number;
+  onOutputResize?: (region: IRectangle) => void;
+  clickHandler?: (event: React.MouseEvent) => void;
+}
+
+export default function Display(props: DisplayProps) {
+  const { VideoService, CustomizationService } = Services;
+
+  const p = {
+    paddingSize: 0,
+    drawUI: false,
+    clickHandler: () => {},
+    onOutputResize: () => {},
+    ...props,
+  };
+
+  const v = useVuex(() => ({
+    paddingColor: CustomizationService.views.displayBackground,
+    baseResolution: VideoService.baseResolution,
+  }));
+
+  const obsDisplay = useRef<OBSDisplay>();
+  const displayEl = useRef<HTMLDivElement>(null);
+
+  useEffect(lifecycle, []);
+
+  function lifecycle() {
+    // componentDidMount
+    createDisplay();
+
+    // componentWillUnmount
+    return function cleanup() {
+      destroyDisplay();
+    };
+  }
+
+  const watchers = useRef({
+    sourceId: p.sourceId,
+    paddingColor: v.paddingColor,
+    baseResolution: v.baseResolution,
+  });
+  if (watchers.current.sourceId !== p.sourceId) {
+    watchers.current.sourceId = p.sourceId;
+    updateDisplay();
+  }
+  if (watchers.current.paddingColor !== v.paddingColor) {
+    watchers.current.paddingColor = v.paddingColor;
+    updateDisplay();
+  }
+  if (watchers.current.baseResolution !== v.baseResolution) {
+    watchers.current.baseResolution = v.baseResolution;
+    if (obsDisplay.current) obsDisplay.current.refreshOutputRegion();
+  }
```

Comment:
React hooks already have the comparison that you've implemented here. You can simplify your code:

```
useEffect(udateDisplay, [p.sourceId, v.paddingColor])
useEffect(refreshOutputRegion, [v.baseResolution])
```

Reply:
dang `useEffect` is even more of a beast than i thought lol